src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,7 +33,11 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        addressValue = address.split(", ");
```

Comment:
Validate `addressValue` as you haven't changed the regex. You can check for 4 comma separated strings to validate.